cudax/include/cuda/experimental/__green_context/green_ctx.cuh

```diff
@@ -0,0 +1,87 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDAX__GREEN_CONTEXT_GREEN_CTX
+#define _CUDAX__GREEN_CONTEXT_GREEN_CTX
+
+#include <cuda/__cccl_config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__cuda/api_wrapper.h>
+
+#include <cuda/experimental/__device/all_devices.cuh>
+#include <cuda/experimental/__utility/driver_api.cuh>
+
+#if CUDART_VERSION >= 12050
+namespace cuda::experimental
+{
+struct device_ref;
+
+struct green_context
```

Comment:
I am not sure where this class will end up, if it will even be named `green_context`. I would prefer to wait with docs to save some work. For now it's just here as a way to test `logical_device`.

We will add a proper implementation and then add documentation as well